driver_test.go

```diff
@@ -338,3 +309,302 @@ func TestIsDdl(t *testing.T) {
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true )`,
+			input:     `UPDATE TestExecContextDml SET key = 100 WHERE key = 1`,
+			wantError: true,
+		},
+		{
+			name: "update nothing",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true ),
+				(3, "three", CAST("three" as bytes), 42, 42, true )`,
+			input: `UPDATE TestExecContextDml SET testSTring = "nothing" WHERE false`,
+			query: `SELECT * FROM TestExecContextDml ORDER BY key`,
+			wantRows: []testExecContextDmlRow{
+				{key: 1, testString: "one", testBytes: []byte("one"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 2, testString: "two", testBytes: []byte("two"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 3, testString: "three", testBytes: []byte("three"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name: "update everything",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true ),
+				(3, "three", CAST("three" as bytes), 42, 42, true )`,
+			input: `UPDATE TestExecContextDml SET testSTring = "everything" WHERE true `,
+			query: `SELECT * FROM TestExecContextDml WHERE testString = "everything"`,
+			wantRows: []testExecContextDmlRow{
+				{key: 1, testString: "everything", testBytes: []byte("one"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 2, testString: "everything", testBytes: []byte("two"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 3, testString: "everything", testBytes: []byte("three"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name:      "update to wrong type",
+			input:     `UPDATE TestExecContextDml SET testBool = 100 WHERE key = 1`,
+			wantError: true,
+		},
+	}
+
+	// Run tests.
+	for _, tc := range tests {
+
+		// Set up test table.
+		if tc.setUp != "" {
+			if _, err = db.ExecContext(ctx, tc.setUp); err != nil {
+				t.Error(err)
+			}
+		}
+
+		// Run DML.
+		_, err = db.ExecContext(ctx, tc.input)
+		if (err != nil) && (!tc.wantError) {
+			t.Errorf("%s: unexpected query error: %v", tc.name, err)
+		}
+		if (err == nil) && (tc.wantError) {
+			t.Errorf("%s: expected query error but error was %v", tc.name, err)
+		}
+
+		// Check rows returned.
+		if tc.query != "" {
+			rows, err := db.QueryContext(ctx, tc.query)
+			if err != nil {
+				t.Error(err)
+			}
+
+			got := []testExecContextDmlRow{}
+			for rows.Next() {
+				var curr testExecContextDmlRow
+				err := rows.Scan(
+					&curr.key, &curr.testString, &curr.testBytes, &curr.testInt, &curr.testFloat, &curr.testBool)
+				if err != nil {
+					t.Error(err)
+				}
+				got = append(got, curr)
+			}
+
+			rows.Close()
+			err = rows.Err()
```

Comment:
This can be merged into the line below. If the value is not used outside of the `if` statement and the line is relatively short, these assignments are generally done in the `if` statement.

Reply:
merged